---
int32[] ids
float32[] x
float32[] y
